docs/src/starknet/index.md

```diff
@@ -51,21 +50,30 @@ response: [0x0]
 > In the above example we supply `sncast` with `--account` and `--url` flags. If `snfoundry.toml` is present, and have these properties set, values provided using these flags will override values from `snfoundry.toml`. Learn more about `snfoundry.toml` configuration [here](../projects/configuration.md#sncast).
 
 
-### Calldata
-
-Some `sncast` commands (namely `call`, `deploy` and `invoke`) allow passing *calldata* - a series of arguments to perform an action with on blockchain.
+### Arguments
 
-In the example above we called a function with an argument: `0x0`, passed using `--calldata` flag.
+Some `sncast` commands (namely `call`, `deploy` and `invoke`) allow passing arguments to perform an action with on the blockchain.
 
-Please note the notation of the argument. The default way of passing calldata is a list of hexadecimally encoded field elements - the *serialized* calldata.
-To obtain the serialized form of the wished data, one must write a Cairo program calling `Serde::serialize` on subsequent arguments and displaying the results.
+Under the hood cast always send request with serialized form of arguments, but it can be passed in 
+human-readable form thanks to the [calldata transformation](./calldata-transformation.md) feature present in Cast.
 
-It is also possible to pass calldata in more friendly, human readable form thanks to the [calldata transformation](./calldata-transformation.md) feature present in Cast.
+In the example above we called a function with a deserialized argument: `'"Charizard"'`, passed using `--arguments` flag.
 
 > ⚠️ **Warning**
 > Cast will not verify the serialized calldata. Any errors caused by passing improper calldata in a serialized form will originate from the network.
 > Basic static analysis is possible only when passing expressions - see [calldata transformation](./calldata-transformation.md).
 
+
+### Using Serialized Calldata
+
+The same result can be achieved by passing serialized calldata, which is a list of hexadecimal-encoded field elements.
+
+For example, this is equivalent to using the --calldata option with the following value: 0x0 0x43686172697a617264  0x9.
+
+To obtain the serialized form of the wished data, you can write a Cairo program that calls `Serde::serialize` on subsequent arguments and displays the results.
```

Comment:
Kind of resolves https://github.com/foundry-rs/starknet-foundry/issues/2650 but I think we might want to expand this in a separate PR anyway.